src/engine/ratecontrol.cpp

```diff
@@ -41,12 +41,29 @@ RateControl::RateControl(QString group,
       m_dRateTempRampbackChange(0.0) {
     m_pScratchController = new PositionScratchController(group);
 
+    // This is the resulting rate ratio that can used for dispaly or calculations.
+    // The track original rate ratio is 1.
+    m_pRateRatio = new ControlObject(ConfigKey(group, "rate_ratio"));
+    connect(m_pRateRatio, SIGNAL(valueChanged(double)),
+            this, SLOT(slotRateRatioChanged(double)),
+            Qt::DirectConnection);
+
     m_pRateDir = new ControlObject(ConfigKey(group, "rate_dir"));
-    m_pRateRange = new ControlPotmeter(ConfigKey(group, "rateRange"), 0.01, 0.90);
+    connect(m_pRateDir, SIGNAL(valueChanged(double)),
+            this, SLOT(slotRateSliderChanged(double)),
+            Qt::DirectConnection);
+    m_pRateRange = new ControlPotmeter(
+            ConfigKey(group, "rateRange"), 0.01, 0.90);
```

Comment:
Change this to 0.01, 4.00 please, in keeping with #1662.

Reply:
Sure, this will happen automatically when merging 2.1 to master.